lib/ZumoOta/src/BootloaderCom.cpp

```diff
@@ -95,28 +105,241 @@ class IntProgCmdProvider: public CmdProvider {
-    static const ResponseInfo m_responses[SEQ_LENGHT]; /**< Array of response information */
+    static  CommandInfo m_cmds[SEQ_LENGHT]; /**< Array of command information */
+    static  ResponseInfo m_responses[SEQ_LENGHT]; /**< Array of response information */
 };
 
-#if 0
+/**
+ * @class FwProgCmdProvider
+ * @brief Implementation of the command provider for firmware programming.
+ *
+ * This class provides commands for firmware programming, specifically tailored
+ * for the Zumo32U4 device.
+ */
 class FwProgCmdProvider : public CmdProvider {
     public:
-        FwProgCmdProvider(FileHandle * fs);
+        /**
+         * @brief Constructor for the FwProgCmdProvider class.
+         * @param fileName The name of the file to be used for firmware programming.
+         * Initializes a new instance of the FwProgCmdProvider class with the provided file name.
+         * @remarks The file specified by fileName should contain the firmware to be programmed.
+         */
+        FwProgCmdProvider(const char* fileName):
+            m_fileName(fileName),
+            m_firmwareBytesRead(0),
+            m_updatedCmd(),
+            m_firmwareBuffer(),
+            m_firmwareFile(),
+            m_updatedCmdBuffer{},
+            m_currWriteMemAddr(0x0000)
+        {}
+
+    /**
+     * @brief Resets the command provider to its initial state.
+     * This method resets the internal state of the command provider.
+     * It checks if the firmware file is open and closes it if necessary.
+     * Then, it attempts to open the firmware file with read access.
+     * @return True if the firmware file is successfully opened, false otherwise.
+     */
+    virtual bool reset()
+    {
+        /*Check if the file is already open.*/
+        if (m_firmwareFile)
+        {
+            m_firmwareFile.close();
+        }
+
+        /*If the file is not yet open, open it.*/
+        m_firmwareFile = LittleFS.open(m_fileName, "r");
+        return false != m_firmwareFile;
 }
-#endif
+
+    /**
+     * @brief Retrieves the next command and the next response.
+     *
+     * This method returns the next command and its associated response
+     * from the command sequence memory.
+     * @param cmd Reference to a pointer to the next command.
+     * @param rsp Reference to a pointer to the next response.
+     * @return True if there is a next command and next response, false otherwise.
+     */
+    virtual bool next(const CommandInfo*& cmd, const ResponseInfo*& rsp)
+    {
+        if (m_nextCmd == FLSPRG_COMPLETE) 
+        {
+            cmd = nullptr;
+            rsp = nullptr;
+
+            if (m_firmwareFile)
+            {
+                m_firmwareFile.close(); // Close the file
+            }
+
+            return false;
+        }
+        else
+        {
+            rsp = &m_responses[m_nextCmd];
+    
+            /*Update the command with required content to write in the flash memory.*/
+            if(true == updateFwProgCmd(m_updatedCmd))

```

Comment:
add blank after if

Reply:
It will be taken care of by the next commit.